### PR TITLE
Added documentation for the new GCP stackdriver scaler

### PR DESCRIPTION
**Pull request by RamCohen**
Added documentation for the new GCP stackdriver scaler

Relates to https://github.com/kedacore/keda/issues/2661

Signed-off-by: Ram Cohen <ram.cohen@gmail.com>

### CODE REVIEW

File: content/docs/2.7/scalers/gcp-stackdriver.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+This specification describes the `gcp-stackdriver` trigger for GCP Stackdriver. It scales based on a metric obtained from issuing a query to Stackdriver.
+
+```yaml
+triggers:
+- type: gcp-stackdriver
+  metadata:
+    projectId: my-project-id
+    filter: 'metric.type="storage.googleapis.com/network/received_bytes_count" AND resource.type="gcs_bucket" AND metric.label.method="WriteObject" AND resource.label.bucket_name="my-gcp-bucket"'
+    metricName: my-gcp-bucket
```
Comment (zroubalik) [inline content/docs/2.7/scalers/gcp-stackdriver.md:20]: Can we please not force users to specify `metricName`, we should generate it instead.

File: content/docs/2.7/scalers/gcp-stackdriver.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+**Parameter list:**
+
+- `projectId` - GCP project Id that contains the metric.
+- `filter` - The stackdriver query filter for obtaining the metric. The metric is for the last minute and if multiple values are returned, the first one is used.
+- `metricName` - Metric name for reporting the value to HPA, prefixed with `s{triggerIndex}-gcp-stackdriver-`.
+- `targetValue` - Average target value to trigger scaling actions. (Default: `5`, Optional)
+
+You can provide in the metadata either `credentialsFromEnv` or `credentialsFromEnvFile`.
+- `credentialsFromEnv` - Set to the name of the environment variable that holds the credential information.
+- `credentialsFromEnvFile` - Set to the name of a json file that holds the credential information.
```
Comment (zroubalik) [inline content/docs/2.7/scalers/gcp-stackdriver.md:35]: I don't see this anywhere used. What is the purpose anyway?

File: content/docs/2.7/scalers/gcp-stackdriver.md (hunk 1 above)
Comment (RamCohen) [reply content/docs/2.7/scalers/gcp-stackdriver.md:20]: I know but there isn't any other parameter I can deduce a reasonable metric from. project_id is too generic and can collide with another such scaler

File: content/docs/2.7/scalers/gcp-stackdriver.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    credentialsFromEnv: GOOGLE_APPLICATION_CREDENTIALS_JSON
+```
+
+**Parameter list:**
+
+- `projectId` - GCP project Id that contains the metric.
+- `filter` - The stackdriver query filter for obtaining the metric. The metric is for the last minute and if multiple values are returned, the first one is used.
+- `targetValue` - Average target value to trigger scaling actions. (Default: `5`, Optional)
+
+The metric name will be generated automatically based on the trigger index and `projectId`, for example: **s0-gcp-stackdriver-myProject**.
```
Comment (zroubalik) [inline content/docs/2.7/scalers/gcp-stackdriver.md:30]: I think you can omit this line, it is just internal detail. Not user facing.

**Review by zroubalik — APPROVED** [review]
LGTM

/hold for others before merging.

**Review by JorTurFer — APPROVED** [review]
LGTM!
Thanks for this! ❤️

**Pull request merged**